File: app/src/main/java/co/edu/uniautonoma/posgradosapp/Retrofit/PeticionesApi.java
package co.edu.uniautonoma.posgradosapp.Retrofit;

import java.util.List;

import co.edu.uniautonoma.posgradosapp.Modelos.Calificaciones;
import co.edu.uniautonoma.posgradosapp.Modelos.Docentes;
import co.edu.uniautonoma.posgradosapp.Modelos.Escuela;
import co.edu.uniautonoma.posgradosapp.Modelos.Horarios;
import co.edu.uniautonoma.posgradosapp.Modelos.Modulos;
import co.edu.uniautonoma.posgradosapp.Modelos.Posgrados;
import co.edu.uniautonoma.posgradosapp.Modelos.Usuarios;
import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface PeticionesApi {

    @GET("get_escuela_.php")
    Observable<Escuela> getEscuela(@Query("id") int id);

    @GET("get_usuario_.php")
    Observable<Usuarios> getUsuario(@Query("correo") String correo);

    @GET("get_all_posgrados_.php")
    Observable<List<Posgrados>> getAllPosgrados();

    @GET("get_all_modulos_.php")
    Observable<List<Modulos>> getAllModulos(@Query("id_posgrado") String id_posgrado);

    @GET("get_all_docentes_.php")
    Observable<List<Docentes>> getAllDocentes(@Query("id_posgrado") String id_posgrado);

    @GET("get_posgrado_.php")
    Observable<Posgrados> getPosgrado(@Query("id_posgrado") String id_posgrado);

    @GET("get_some_modulos_.php")
    Observable<List<Modulos>> getSomeModulos(@Query("id_posgrado") String id_posgrado, @Query("semestre") int semestre);

    @GET("get_some_docentes_.php")
    Observable<List<Docentes>> getSomeDocentes(@Query("id_posgrado") String id_posgrado, @Query("semestre") int semestre);

    @GET("get_horario_.php")
    Observable<List<Horarios>> getHorario(@Query("id_posgrado") String id_posgrado, @Query("semestre") int semestre);

    @GET("get_some_calificaciones_.php")
    Observable<List<Calificaciones>> getSomeCalificaciones(@Query("id_posgrado") String id_posgrado, @Query("semestre") int semestre, @Query("id_usuario") String id_usuario);

    @GET("get_some_calificacion_.php")
    Observable<Calificaciones> getSomeCalificacion(@Query("id_usuario") String id_usuario, @Query("id_docente") String id_docente);

    @FormUrlEncoded
    @POST("update_calificacion.php")
    Observable<Calificaciones> updateCalificacion(@Field("calificacion") float calificacion, @Field("id_usuario") String id_usuario, @Field("id_docente") String id_docente);

    @FormUrlEncoded
    @POST("add_calificacion.php")
    Observable<Calificaciones> addCalificacion(@Field("calificacion") float calificacion, @Field("id_usuario") String id_usuario, @Field("id_docente") String id_docente);
}

File: data/src/main/java/co/edu/uniautonoma/posgradosapp/data/datasource/remote/retrofit/PeticionesApi.kt
package co.edu.uniautonoma.posgradosapp.data.datasource.remote.retrofit

import co.edu.uniautonoma.posgradosapp.data.models.*
import retrofit2.Response
import retrofit2.http.*

interface PeticionesApi {
    @GET("get_escuela_.php")
    suspend fun getEscuela(@Query("id") id: Int): Response<EscuelaApi?>

    @GET("get_usuario_.php")
    suspend fun getUsuario(@Query("correo") correo: String?): Response<UsuarioApi?>

    @GET("get_all_posgrados_.php")
    suspend fun getAllPosgrados(): Response<List<PosgradosApi>?>

    @GET("get_all_modulos_.php")
    suspend fun getAllModulos(@Query("id_posgrado") id_posgrado: String?): Response<List<ModulosApi>?>

    @GET("get_all_docentes_.php")
    suspend fun getAllDocentes(@Query("id_posgrado") id_posgrado: String?): Response<List<DocentesApi>?>

    @GET("get_posgrado_.php")
    suspend fun getPosgrado(@Query("id_posgrado") id_posgrado: String?): Response<PosgradosApi?>

    @GET("get_some_modulos_.php")
    suspend fun getSomeModulos(@Query("id_posgrado") id_posgrado: String?, @Query("semestre") semestre: Int): Response<List<ModulosApi>?>

    @GET("get_some_docentes_.php")
    suspend fun getSomeDocentes(@Query("id_posgrado") id_posgrado: String?, @Query("semestre") semestre: Int): Response<List<DocentesApi>?>

    @GET("get_horario_.php")
    suspend fun getHorario(@Query("id_posgrado") id_posgrado: String?, @Query("semestre") semestre: Int): Response<List<HorariosApi>?>

    @GET("get_some_calificaciones_.php")
    suspend fun getSomeCalificaciones(@Query("id_posgrado") id_posgrado: String?, @Query("semestre") semestre: Int, @Query("id_usuario") id_usuario: String?): Response<List<CalificacionesApi>?>

    @GET("get_some_calificacion_.php")
    suspend fun getSomeCalificacion(@Query("id_usuario") id_usuario: String?, @Query("id_docente") id_docente: String?): Response<CalificacionesApi?>

    @FormUrlEncoded
    @POST("update_calificacion.php")
    suspend fun updateCalificacion(@Field("calificacion") calificacion: Float, @Field("id_usuario") id_usuario: String?, @Field("id_docente") id_docente: String?): Response<CalificacionesApi?>

    @FormUrlEncoded
    @POST("add_calificacion.php")
    suspend fun addCalificacion(@Field("calificacion") calificacion: Float, @Field("id_usuario") id_usuario: String?, @Field("id_docente") id_docente: String?): Response<CalificacionesApi?>
}